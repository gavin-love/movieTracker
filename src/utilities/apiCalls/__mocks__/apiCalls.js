export const submitFavorite = jest.fn().mockImplementation(() => Promise.resolve());
export const removeFavorite = jest.fn().mockImplementation(() => Promise.resolve());
export const postNewUser = jest.fn().mockImplementation(() => Promise.resolve( {
  id: 1
}));
export const getUser = jest.fn().mockImplementation(() => Promise.resolve({
  data: {
    id: 1,
    name: 'tim',
    password: 'password',
    email: 'email'
  }
}))