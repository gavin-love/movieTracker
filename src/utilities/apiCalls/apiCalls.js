import { apiKey } from "../../apiKeys";
import { cleanMovies } from "../cleaner/cleaner";

export const movieFetch = async () => {
  const url = `https://api.themoviedb.org/3/discover/movie?api_key=${apiKey}&primary_release_date.gte=2014-09-15&primary_release_date.lte=2014-10-22`;

  const response = await fetch(url);
  const movieData = await response.json();
  const movieResults = movieData.results;
  return cleanMovies(movieResults);
};

export const submitFavorite = async (favorite, user) => {
  const url = "http://localhost:3000/api/users/favorites/new";
  const options = {
    method: "POST",
    body: JSON.stringify({ user_id: user.user_id, ...favorite }),
    headers: { "Content-Type": "application/json" }
  };
  const response = await fetch(url, options);
};

export const removeFavorite = async favorite => {
  const url = `http://localhost:3000/api/users/${favorite.user_id}/favorites/${
    favorite.movie_id
  }`;
  const options = {
    method: "DELETE"
  };

  const response = await fetch(url, options);
};

export const postNewUser = async user => {
  const url = "http://localhost:3000/api/users/new";
  const options = {
    method: "POST",
    body: JSON.stringify(user),
    headers: {
      Accept: "application/json",
      "Content-Type": "application/json"
    }
  };
  const userFetch = await fetch(url, options);
  const response = await userFetch.json();
  return response;
};

export const getUser = async user => {
  const url = "http://localhost:3000/api/users";
  const email = user.email.toLowerCase();
  const userOptions = {
    method: "POST",
    body: JSON.stringify({ email, password: user.password }),
    headers: {
      Accept: "application/json",
      "Content-Type": "application/json"
    }
  };
  const response = await fetch(url, userOptions);
  const data = await response.json();
  return data;
};
