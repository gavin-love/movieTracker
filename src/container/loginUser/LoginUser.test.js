import React from 'react';
import { shallow } from 'enzyme';
import { LoginUser, mapDispatchToProps } from './LoginUser';
import { postNewUser } from '../../utilities/apiCalls/apiCalls';

jest.mock('../../utilities/apiCalls/apiCalls')

describe('LoginUser', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<LoginUser />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it.skip('should have an initial state', () => {
    const expected = {
      name: '',
      email: '', 
      password: ''
    }
    expect(wrapper.state()).toEqual(expected);
  })

describe('handleChange', () => {
  it.skip('should update state', () => {
    let mockEvent = {
      target: {
        name: 'name',
        value: 'cat'
      }
    }
    const expected = {
        name: 'cat',
        email: '', 
        password: ''
      }
    wrapper.instance().handleChange(mockEvent);
    expect(wrapper.state()).toEqual(expected);
    })
  })

  describe('handleSubmit', () => {
    it.skip('should call postNewUser with the correct params', () => {
      const mockEvent = {
        preventDefault: jest.fn()
      }
      Promise.resolve(wrapper.instance().handleSubmit(mockEvent));
      expect(postNewUser).toHaveBeenCalledWith(wrapper.state());
    })
    it.skip('should call handleCreateUser with the correct params', async () => {
      const mockEvent = {
        preventDefault: jest.fn()
      }

      const mockUser = {
        id: 1
      }
      await wrapper.instance().handleSubmit(mockEvent);
      expect(mockHandleCreateUser).toHaveBeenCalledWith(mockUser);
    })
    it.skip('should call clearError', async () => {
      const mockEvent = {
        preventDefault: jest.fn()
      } 
      await wrapper.instance().handleSubmit(mockEvent);
      expect(mockHandleError).toHaveBeenCalled();
    })
  })

  describe('mapDispatchToProps', () => {
    it.skip('should call handleCreateUser with the right params', () => {
      const mockDispatch = jest.fn();
      const mappedProps = mapDispatchToProps(mockDispatch);
      const mockAction = {
        type: 'ADD_USER', 
        user_id: 4
      }
      const mockUser = {
        id: 4
      }
      mappedProps.handleCreateUser(mockUser);
      expect(mockDispatch).toHaveBeenCalledWith(mockAction);
    })

    it.skip('should call handleError with the right params', () => {
      const mockDispatch = jest.fn();
      const mappedProps = mapDispatchToProps(mockDispatch);
      const mockAction = {
        type: 'ADD_ERROR', 
        error: 'error'
      }
      const mockError = 'error';

      mappedProps.handleError(mockError);
      expect(mockDispatch).toHaveBeenCalledWith(mockAction);
    })
  })

    it.skip('should call clearError with the right params', () => {
      const mockDispatch = jest.fn();
      const mappedProps = mapDispatchToProps(mockDispatch);
      const mockAction = {
        type: 'CLEAR_ERROR', 
      }
      mappedProps.clearError();
      expect(mockDispatch).toHaveBeenCalled();
    })
  })