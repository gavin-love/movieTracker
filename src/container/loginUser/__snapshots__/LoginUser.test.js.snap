// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginUser should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LoginUser />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h5
          className="login_title"
>
          Account Login
</h5>,
        <input
          className="login_user"
          name="email"
          onChange={[Function]}
          placeholder="Email"
          type="text"
          value=""
/>,
        <input
          className="login_user"
          name="password"
          onChange={[Function]}
          placeholder="Password"
          type="password"
          value=""
/>,
        <button
          className="login_button"
>
          Submit
</button>,
      ],
      "className": "log_in",
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Account Login",
          "className": "login_title",
        },
        "ref": null,
        "rendered": "Account Login",
        "type": "h5",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "login_user",
          "name": "email",
          "onChange": [Function],
          "placeholder": "Email",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "login_user",
          "name": "password",
          "onChange": [Function],
          "placeholder": "Password",
          "type": "password",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Submit",
          "className": "login_button",
        },
        "ref": null,
        "rendered": "Submit",
        "type": "button",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h5
            className="login_title"
>
            Account Login
</h5>,
          <input
            className="login_user"
            name="email"
            onChange={[Function]}
            placeholder="Email"
            type="text"
            value=""
/>,
          <input
            className="login_user"
            name="password"
            onChange={[Function]}
            placeholder="Password"
            type="password"
            value=""
/>,
          <button
            className="login_button"
>
            Submit
</button>,
        ],
        "className": "log_in",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Account Login",
            "className": "login_title",
          },
          "ref": null,
          "rendered": "Account Login",
          "type": "h5",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "login_user",
            "name": "email",
            "onChange": [Function],
            "placeholder": "Email",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "login_user",
            "name": "password",
            "onChange": [Function],
            "placeholder": "Password",
            "type": "password",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Submit",
            "className": "login_button",
          },
          "ref": null,
          "rendered": "Submit",
          "type": "button",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
